### PR TITLE
Clean up unused code according to coverage report

### PR DESCRIPTION
**Pull request by lialan**

### CODE REVIEW

**Comment by zksync-admin-bot2** [comment]
Benchmark results:
```
╔═╡ Size (-%) ╞═════════════════════╡ All ╞═╗
║ Mean                                0.000 ║
║ Best                                0.000 ║
║ Worst                               0.000 ║
║ Total                               0.000 ║
╠═╡ Cycles (-%) ╞═══════════════════╡ All ╞═╣
║ Mean                                0.000 ║
║ Best                                0.000 ║
║ Worst                               0.000 ║
║ Total                               0.000 ║
╚═══════════════════════════════════════════╝

╔═╡ Size (-%) ╞═══════════════╡ Real life ╞═╗
║ Mean                                0.000 ║
║ Best                                0.000 ║
║ Worst                               0.000 ║
║ Total                               0.000 ║
╠═╡ Cycles (-%) ╞═════════════╡ Real life ╞═╣
║ Mean                                0.000 ║
║ Best                                0.000 ║
║ Worst                               0.000 ║
║ Total                               0.000 ║
╚═══════════════════════════════════════════╝

```

**Comment by zksync-admin-bot2** [comment]
Benchmark results:
```
╔═╡ Size (-%) ╞═════════════════════╡ All ╞═╗
║ Mean                                0.000 ║
║ Best                                0.000 ║
║ Worst                               0.000 ║
║ Total                               0.000 ║
╠═╡ Cycles (-%) ╞═══════════════════╡ All ╞═╣
║ Mean                                0.000 ║
║ Best                                0.000 ║
║ Worst                               0.000 ║
║ Total                               0.000 ║
╚═══════════════════════════════════════════╝

╔═╡ Size (-%) ╞═══════════════╡ Real life ╞═╗
║ Mean                                0.000 ║
║ Best                                0.000 ║
║ Worst                               0.000 ║
║ Total                               0.000 ║
╠═╡ Cycles (-%) ╞═════════════╡ Real life ╞═╣
║ Mean                                0.000 ║
║ Best                                0.000 ║
║ Worst                               0.000 ║
║ Total                               0.000 ║
╚═══════════════════════════════════════════╝

```

File: llvm/lib/Target/SyncVM/SyncVMExpandPseudoInsts.cpp [hunk 1]
```diff
@@ -174,6 +110,17 @@ bool SyncVMExpandPseudo::runOnMachineFunction(MachineFunction &MF) {
           LLVM_DEBUG(dbgs() << "Converting PTR_TO_INT to: "; NewMI->dump());
         }
         PseudoInst.push_back(&MI);
+      } else if (MI.getOpcode() == SyncVM::LOADCONST) {
```
Comment (akiramenai) [inline llvm/lib/Target/SyncVM/SyncVMExpandPseudoInsts.cpp:113]: Could you explain how has it worked before?

File: llvm/lib/Target/SyncVM/SyncVMRegisterInfo.cpp [hunk 2]
```diff
@@ -72,20 +72,14 @@ void SyncVMRegisterInfo::eliminateFrameIndex(MachineBasicBlock::iterator II,
...
-                   .addImm(SyncVMCC::COND_NONE)
-                   .getInstr();
-    else
-      SPInst = BuildMI(MBB, II, DL, TII.get(SyncVM::ADDirr_s))
-                   .addDef(MI.getOperand(0).getReg())
-                   .addImm(Offset /32)
-                   .add(SPInst->getOperand(0))
-                   .addImm(SyncVMCC::COND_NONE)
-                   .getInstr();
+    assert(Offset < 0 && "On SyncVM, offset cannot be positive");
```
Comment (akiramenai) [inline llvm/lib/Target/SyncVM/SyncVMRegisterInfo.cpp:75]: Nothing technically prevent us from having it, i.e. the VM supports. Solidity doesn't produce negative offsets, we don't have much tests to cover negative offsets, etc, but in principle if we (or our partners) ever support C++ or something alike, negative offsets will be of need. So I prefer to keep the original implementation.

**Review by akiramenai — CHANGES_REQUESTED** [review]

File: llvm/lib/Target/SyncVM/SyncVMRegisterInfo.cpp (hunk 2 above)
Comment (lialan) [reply llvm/lib/Target/SyncVM/SyncVMRegisterInfo.cpp:75]: It is actually stated in the spec that stack addressing mode does not allow zero or positive offsets.

But you are right, I found this security feature too strict for backend implementation. We should either update the spec or do this assert to guard this off.

File: llvm/lib/Target/SyncVM/SyncVMExpandPseudoInsts.cpp (hunk 1 above)
Comment (lialan) [reply llvm/lib/Target/SyncVM/SyncVMExpandPseudoInsts.cpp:113]: My previous comments were wrong: I just moved the LOADCONST around to make the cod e more concise.

File: llvm/lib/Target/SyncVM/SyncVMRegisterInfo.cpp (hunk 2 above)
Comment (akiramenai) [reply llvm/lib/Target/SyncVM/SyncVMRegisterInfo.cpp:75]: I missed that we are calculating the offset here based on the frame structure. We indeed should never have it negative or something went wrong with the frame construction. So, let's go with your option.

**Review by akiramenai — APPROVED** [review]
Please squash

**Comment by zksync-admin-bot2** [comment]
Benchmark results:
```
╔═╡ Size (-%) ╞═════════════════════╡ All ╞═╗
║ Mean                                0.000 ║
║ Best                                0.000 ║
║ Worst                               0.000 ║
║ Total                               0.000 ║
╠═╡ Cycles (-%) ╞═══════════════════╡ All ╞═╣
║ Mean                                0.000 ║
║ Best                                0.000 ║
║ Worst                               0.000 ║
║ Total                               0.000 ║
╚═══════════════════════════════════════════╝

╔═╡ Size (-%) ╞═══════════════╡ Real life ╞═╗
║ Mean                                0.000 ║
║ Best                                0.000 ║
║ Worst                               0.000 ║
║ Total                               0.000 ║
╠═╡ Cycles (-%) ╞═════════════╡ Real life ╞═╣
║ Mean                                0.000 ║
║ Best                                0.000 ║
║ Worst                               0.000 ║
║ Total                               0.000 ║
╚═══════════════════════════════════════════╝

```

**Pull request merged**